com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/ModelUtil.java

```diff
@@ -361,79 +363,194 @@ public static ArrayList<String> loadMaterialNames(Scene scene) {
+        }
+        if (inMesh.bones != null) {
+            copyIntArray(inMesh.bones, inIndex, outMesh.bones, outIndex, 4);
+        }
+        if (inMesh.texCoords0 != null) {
+            copyFloatArray(inMesh.texCoords0, inIndex, outMesh.texCoords0, outIndex, inMesh.texCoords0NumComponents);
+        }
+        if (inMesh.texCoords1 != null) {
+            copyFloatArray(inMesh.texCoords1, inIndex, outMesh.texCoords1, outIndex, inMesh.texCoords1NumComponents);
+        }
+    }
+
+    private static void splitMesh(ModelImporter.Mesh inMesh, List<ModelImporter.Mesh> outMeshes) {
+        int triangleCount = inMesh.indexCount / 3;
+        int vertexCount = inMesh.vertexCount;
+
+        int vcount = 0;
+        ModelImporter.Mesh newMesh = null;
+        HashMap<Integer, Integer> oldToNewIndex = null;
+        ArrayList<Integer> newIndices = null;
+
+        for (int i = 0; i < triangleCount; ++i) {
+
+            if (newMesh == null) {
+                oldToNewIndex = new HashMap<Integer, Integer>();
+                newIndices = new ArrayList<Integer>();
+
+                newMesh = new Mesh();
+                newMesh.material = inMesh.material;
+                newMesh.name = String.format("%s_%d", inMesh.name, outMeshes.size());
+
+                newMesh.texCoords0NumComponents = inMesh.texCoords0NumComponents;
+                newMesh.texCoords1NumComponents = inMesh.texCoords1NumComponents;
+
+                if (inMesh.positions != null)
+                    newMesh.positions = new float[MAX_SPLIT_VCOUNT*3];
+                if (inMesh.normals != null)
+                    newMesh.normals = new float[MAX_SPLIT_VCOUNT*3];
+                if (inMesh.tangents != null)
+                    newMesh.tangents = new float[MAX_SPLIT_VCOUNT*3];
+                if (inMesh.colors != null)
+                    newMesh.colors = new float[MAX_SPLIT_VCOUNT * 4];
+                if (inMesh.weights != null)
+                    newMesh.weights = new float[MAX_SPLIT_VCOUNT * 4];
+                if (inMesh.bones != null)
+                    newMesh.bones = new int[MAX_SPLIT_VCOUNT * 4];
+                if (inMesh.texCoords0 != null)
+                    newMesh.texCoords0 = new float[MAX_SPLIT_VCOUNT*3];
+                if (inMesh.texCoords1 != null)
+                    newMesh.texCoords1 = new float[MAX_SPLIT_VCOUNT*3];
+            }
+
+            int index0 = inMesh.indices[i*3+0];
+            int index1 = inMesh.indices[i*3+1];
+            int index2 = inMesh.indices[i*3+2];
+
+            int newIndex0 = oldToNewIndex.getOrDefault(index0, -1);
+            int newIndex1 = oldToNewIndex.getOrDefault(index1, -1);
+            int newIndex2 = oldToNewIndex.getOrDefault(index2, -1);
+
+            if (newIndex0 == -1) {
+                newIndex0 = vcount++;
+                oldToNewIndex.put(index0, newIndex0);
+                copyVertex(inMesh, index0, newMesh, newIndex0);
+            }
```

Comment:
If the index didn't exist in the look up for this new mesh, then we need to add this vertex to the mesh, and add the new index to the table.